### PR TITLE
Fix white handicap adjustment by moving the +/- to the "rank" domain

### PR DESCRIPTION
**Pull request by dexonsmith**
To update rating after a game result, we adjust the opponent's rating by adding or subtracting the effective handicap.

The code before this commit was correct for adjusting black's rating, but incorrect for adjusting white's rating:

- `get_handicap_adjustment` always behaved as-if adjusting black's rating, *adding* the effective handicap.
- The caller passed on `+difference` (black) or `-difference` (white).
    - `-difference` was too small when white's rating was >1500.
    - `-difference` was too big when white's rating was <1500.

With this patch, we do the +/- in the "rank" domain where the effective handicap is symmetric:

- `get_handicap_adjustment` now takes a "black"/"white" parameter so it knows which rating it is adjusting, and returns the correct difference.
- The caller passes on this (maybe negative) difference without inverting it.

As a drive-by, also updated `analysis/analyze_gor.py` to pass the correct parameters for the `expected_win_probability` calculation (now the parameters match the ratings adjustments higher in the same function).

### CODE REVIEW

**Comment by dexonsmith** [comment]
Running the following:
```
% ./analysis/analyze_glicko2_one_game_at_a_time.py --ogs --egf --aga
% ./analysis/analyze_glicko2_one_game_at_a_time.py --handicap-rank-difference-{19x19,small} --ogs --egf --aga
```
I'll attach results when they finish.

**Comment by dexonsmith** [comment]
Attaching old/new of both of these:
```
% ./analysis/analyze_glicko2_one_game_at_a_time.py --ogs --egf --aga
% ./analysis/analyze_glicko2_one_game_at_a_time.py --handicap-rank-difference-{19x19,small} --ogs --egf --aga
```

Before this patch (old):
[old.txt](https://github.com/online-go/goratings/files/13933084/old.txt)
[old-rankdiff.txt](https://github.com/online-go/goratings/files/13933086/old-rankdiff.txt)

With this patch (new):
[new.txt](https://github.com/online-go/goratings/files/13933033/new.txt)
[new-rankdiff.txt](https://github.com/online-go/goratings/files/13933088/new-rankdiff.txt)

File: analysis/analyze_gor.py [hunk 1]
```diff
@@ -69,10 +68,9 @@ def process_game(self, game: GameRecord) -> GorAnalytics:
         return GorAnalytics(
             skipped=False,
             game=game,
-            expected_win_rate=black.with_handicap(get_handicap_adjustment(white.rating, game.handicap,
+            expected_win_rate=black.with_handicap(get_handicap_adjustment("black", black.rating, game.handicap,
```
Comment (dexonsmith) [inline analysis/analyze_gor.py:71]: This is the drive-by code change. Now it matches the first call to `gor_update()` above, which I'm 99.9% is the intent (since every either file does this).

**Comment by anoek** [comment]
Good catch, thanks for the fix!
